storage/interface.go

```diff
@@ -180,6 +180,12 @@ type Appender interface {
 	ExemplarAppender
 }
 
+// OptionalGetRef is used by downstream projects (e.g. Cortex) to avoid maintaining a parallel set of references.
+type OptionalGetRef interface {
+	// Returns reference number that can be used to pass to Appender.Append()
+	GetRef(lset labels.Labels) (uint64, bool)
```

Comment:
Good question about the 0 on wrong case. I think bool might more specific so I would leave it as is. We can iterate over it later.

Reply:
Regarding how we manage this elsewhere, 0 is enough.